### PR TITLE
Feature/setup apis

### PR DESCRIPTION
**Pull request by ericahagle**
# Description
- Adds fetch for all movie data
  - Also added a check in `Movies` to display "Loading..." if the movies haven't loaded in yet, to make the UX nicer
- Removes unnecessary imports for `mock-data` and `Card` in `App.js`
  - I left the `mock-data.js` file in the repo for now, in case we get crazy later and decide that we want to figure out how to do some unit tests
- Also includes some minor formatting tweaks to the code

# Notes
- Basic error handling is in, but we can work on handling it more elegantly in a subsequent PR for this ticket:
https://github.com/rjsturing/rancid-tomatillos/issues/13

# Checklist
- [x] My PR has an appropriately descriptive and concise title.
- [x] My code follows the Turing Style Guides and best practices.
- [x] I ran the code locally and verified that there are no visible errors.
- [x] feat: My PR clearly describes what feature I'm adding and any changes needed to make it work.
- [x] refactor: My PR clearly describes what code I'm refactoring and why.

### CODE REVIEW

**Comment by rjsturing** [comment]
Great work, fetch is functioning! 🎉 Next up is error handling.

**Review by rjsturing — APPROVED** [review]
Great work, fetch is functioning! 🎉 Next up is error handling.

File: src/App/App.js [hunk 1]
```diff
@@ -1,16 +1,26 @@
+import { useState, useEffect } from "react";
 import "./App.css";
 import Movies from "../Movies/Movies";
 import MovieDetail from "../MovieDetail/MovieDetail";
-import Card from "../Card/Card";
-import movieData from "../mock-data";
-import { useState, useEffect } from "react";
 
 function App() {
+  const [movies, setMovies] = useState(null);
```
Comment (nicktu12) [inline src/App/App.js:7]: State should represent the eventual data type, in this case I would think empty array

File: src/Movies/Movies.js [hunk 2]
```diff
@@ -2,6 +2,10 @@ import './Movies.css';
 import Card from '../Card/Card';
 
 function Movies({ movies, selectMovie }) {
+  if (!Array.isArray(movies) || movies.length === 0) {
```
Comment (nicktu12) [inline src/Movies/Movies.js:5]: See above comment about using empty array for state

File: src/App/App.js [hunk 3]
```diff
@@ -1,16 +1,26 @@
...
   const [selectedMovie, setSelectedMovie] = useState(null);
 
+  const getAllMovies = () => {
+    fetch('https://rancid-tomatillos.herokuapp.com/api/v2/movies')
```
Comment (nicktu12) [inline src/App/App.js:11]: Consider refactoring to a separate API calls file for separation of concerns!

File: src/Movies/Movies.js (hunk 2 above)
Comment (ericahagle) [reply src/Movies/Movies.js:5]: Ditto.

File: src/App/App.js (hunk 3 above)
Comment (ericahagle) [reply src/App/App.js:11]: @nicktu12 To clarify, you mean to define this fetch in an `apiCalls.js` doc (or some such) and then invoke it here in this file. Is that right?

File: src/Movies/Movies.js (hunk 2 above)
Comment (ericahagle) [reply src/Movies/Movies.js:5]: @nicktu12 Actually, a question on this—I'm not sure what to change here? We're saying that if it's not the movies array OR if the movies array has no movies in it, we show "Loading..." on the screen to alert the user that it hasn't loaded yet... what would we want to do differently?

Comment (nicktu12) [reply src/Movies/Movies.js:5]: The variable movies should _always_ be an array - the way its setup now is it is actually 2 data types, either an array or null. Typescript would not allow this.

Comment (nicktu12) [reply src/Movies/Movies.js:5]: all that to say the first part of this logical operator should be removed, cause movies should _**always**_ be an array

Comment (ericahagle) [reply src/Movies/Movies.js:5]: Ah ok, I see what you mean. Thanks, @nicktu12 !

**Pull request merged**